File: app/src/main/java/paul/barthuel/go4lunch/ui/workmates/WorkmatesInfoDiffCallBack.java
package paul.barthuel.go4lunch.ui.workmates;

import androidx.annotation.NonNull;

class WorkmatesInfoDiffCallBack extends androidx.recyclerview.widget.DiffUtil.ItemCallback<WorkmatesInfo> {

    @Override
    public boolean areItemsTheSame(@NonNull WorkmatesInfo oldItem, @NonNull WorkmatesInfo newItem) {
        return oldItem.getName().equals(newItem.getName());
    }

    @Override
    public boolean areContentsTheSame(@NonNull WorkmatesInfo oldItem, @NonNull WorkmatesInfo newItem) {
        return oldItem.getName().equals(newItem.getName())
                && oldItem.getImage().equals(newItem.getImage());
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/ui/workmates/WorkmatesInfoDiffCallBack.kt
package paul.barthuel.go4lunch.ui.workmates

import androidx.recyclerview.widget.DiffUtil

internal class WorkmatesInfoDiffCallBack : DiffUtil.ItemCallback<WorkmatesInfo>() {
    override fun areItemsTheSame(oldItem: WorkmatesInfo, newItem: WorkmatesInfo): Boolean {
        return oldItem.name == newItem.name
    }

    override fun areContentsTheSame(oldItem: WorkmatesInfo, newItem: WorkmatesInfo): Boolean {
        return oldItem.name == newItem.name && oldItem.image == newItem.image
    }
}